modin/core/io/text/csv_glob_dispatcher.py

```diff
@@ -348,14 +345,27 @@ def file_exists(cls, file_path: str, storage_options=None) -> bool:
                 else:
                     new_storage_options = {}
 
-                s3fs = S3FS.S3FileSystem(anon=False, **new_storage_options)
+                fs, _ = fsspec.core.url_to_fs(file_path, **new_storage_options)
                 exists = False
                 try:
-                    exists = len(s3fs.glob(file_path)) > 0 or exists
-                except (NoCredentialsError, PermissionError, EndpointConnectionError):
+                    exists = fs.exists(file_path)
+                except (
+                    NoCredentialsError,
+                    PermissionError,
+                    EndpointConnectionError,
+                    ConnectTimeoutError,
+                ):
                     pass
-                s3fs = S3FS.S3FileSystem(anon=True, **new_storage_options)
-                return exists or len(s3fs.glob(file_path)) > 0
+                fs, _ = fsspec.core.url_to_fs(
+                    file_path, anon=True, **new_storage_options
+                )
+                return exists or len(fs.glob(file_path)) > 0
+
+            if fsspec.core.split_protocol(file_path)[0] in ("http", "https"):
```

Comment:
why it's not a condition like `if proto not in ("s3", "S3")`, what if I specify a URL like `ftp://foo/bar.csv`?..

Reply:
See: https://github.com/modin-project/modin/issues/4742

Reply:
Condition was changed to `if is_url(file_path):` to throw exceptions on each URL type